keras_cv/models/vgg16.py

```diff
@@ -73,93 +101,135 @@ def VGG16(
-
-    # Block 5
-    x = layers.Conv2D(
-        512, (3, 3), activation="relu", padding="same", name="block5_conv1"
-    )(x)
-    x = layers.Conv2D(
-        512, (3, 3), activation="relu", padding="same", name="block5_conv2"
-    )(x)
-    x = layers.Conv2D(
-        512, (3, 3), activation="relu", padding="same", name="block5_conv3"
-    )(x)
-    x = layers.MaxPooling2D((2, 2), strides=(2, 2), name="block5_pool")(x)
-
-    if include_top:
-        x = layers.Flatten(name="flatten")(x)
-        x = layers.Dense(4096, activation="relu", name="fc1")(x)
-        x = layers.Dense(4096, activation="relu", name="fc2")(x)
-        x = layers.Dense(
-            classes, activation=classifier_activation, name="predictions"
-        )(x)
-    else:
-        if pooling == "avg":
-            x = layers.GlobalAveragePooling2D()(x)
-        elif pooling == "max":
-            x = layers.GlobalMaxPooling2D()(x)
+        x = build_vgg_block(
+            input_tensor=x,
+            num_layers=3,
+            kernel_size=256,
+            stride=(3, 3),
+            activation="relu",
+            padding="same",
+            max_pool=True,
+            name="block3",
+        )
 
-    model = keras.Model(inputs, x, name=name, **kwargs)
+        x = build_vgg_block(
+            input_tensor=x,
+            num_layers=3,
+            kernel_size=512,
+            stride=(3, 3),
+            activation="relu",
+            padding="same",
+            max_pool=True,
+            name="block4",
+        )
+
+        x = build_vgg_block(
+            input_tensor=x,
+            num_layers=3,
+            kernel_size=512,
+            stride=(3, 3),
+            activation="relu",
+            padding="same",
+            max_pool=True,
+            name="block5",
+        )
 
-    if weights is not None:
-        model.load_weights(weights)
-    return model
+        if include_top:
+            x = layers.Flatten(name="flatten")(x)
+            x = layers.Dense(4096, activation="relu", name="fc1")(x)
+            x = layers.Dense(4096, activation="relu", name="fc2")(x)
+            x = layers.Dense(
+                classes, activation=classifier_activation, name="predictions"
+            )(x)
+        else:
+            if pooling == "avg":
+                x = layers.GlobalAveragePooling2D()(x)
+            elif pooling == "max":
+                x = layers.GlobalMaxPooling2D()(x)
+
+        super().__init__(inputs=inputs, outputs=x, name=name, **kwargs)
+        if weights is not None:
+            self.load_weights(weights)
+
+    def get_config(self):
+        return {
+            "include_rescaling": self.include_rescaling,
+            "include_top": self.include_top,
+            "name": self.name,
+            "input_shape": self.input_shape[1:],
+            "input_tensor": self.input_tensor,
+            "pooling": self.pooling,
+            "classes": self.classes,
+            "classifier_activation": self.classifier_activation,
+        }
```

Comment:
Please include:
`"trainable": self.trainable` 
as well

Reply:
I have done all the changes. Kindly please review and let me know if any changes to be made.